packages/sdk-auth/src/tokenProvider.js

```diff
@@ -79,6 +79,8 @@ export default class TokenProvider {
     return this.fetchTokenInfoPromise.then((tokenInfo) => {
       this.fetchTokenInfoPromise = null
       return tokenInfo
+    }).catch(() => {
+        this.fetchTokenInfoPromise = null
```

Comment:
```suggestion
    }).catch((e) => {
        this.fetchTokenInfoPromise = null
        throw e
```
Just swallowing the error looks dangerous to me, how about above?

Reply:
Didn't realize and even the resetting the promise so changed the behaviour